test/plugin_functional/plugins/core_plugin_a/server/plugin.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Plugin, CoreSetup } from 'kibana/server';
+
+export interface PluginARequestContext {
+  ping: () => Promise<string>;
+}
+
+declare module 'kibana/server' {
+  interface RequestHandlerContext {
+    pluginA?: PluginARequestContext;
+  }
+}
+
+export class CorePluginAPlugin implements Plugin {
+  public setup(core: CoreSetup, deps: {}) {
+    core.http.registerRouteHandlerContext('pluginA', context => {
+      return {
+        ping: () =>
+          context.core!.elasticsearch.adminClient.callAsInternalUser('ping') as Promise<string>,
```

Comment:
TODO later: core context is always defined when plugin is in setup phase

Reply:
This is because of this type? https://github.com/elastic/kibana/blob/master/src/core/utils/context.ts#L42

I was a little conflicted on how this should work since plugins can also create ContextContainers, it's not clear that `core` will always be present. The only idea I have is adding an additional type parameter that defines which `keyof TContext` are always present.